inter/block.go

```diff
@@ -15,12 +15,31 @@ type Block struct {
 	SkippedTxs  []uint32      // indexes of skipped txs, starting from first tx of first event, ending with last tx of last event
 	GasUsed     uint64
 	Root        hash.Hash
+	prevRandao  common.Hash
 }
 
 func (b *Block) EstimateSize() int {
 	return (len(b.Events)+len(b.InternalTxs)+len(b.Txs)+1+1)*32 + len(b.SkippedTxs)*4 + 8 + 8
 }
 
+// GetPrevRandao returns the PrevRandao for current block.
+func (b *Block) GetPrevRandao() common.Hash {
+	if b.prevRandao == (common.Hash{}) {
+		b.computePrevRandao()
+	}
+	return b.prevRandao
+}
+
+// computePrevRandao computes the PrevRandao from transaction hashes.
+func (b *Block) computePrevRandao() {
+	for _, event := range b.Events {
```

Comment:
You should reset `b.prevRandao` to zero before computing the xor of the events (conservative coding).

Reply:
I dont think its necessary because this func gets called only if `prevrandao` is zero.
```// GetPrevRandao returns the PrevRandao for current block.
func (b *Block) GetPrevRandao() common.Hash {
	if b.prevRandao == (common.Hash{}) {
		b.computePrevRandao()
	}
	return b.prevRandao
}

But yeah, in future we might call this func somewhere else so its a good practice.